### PR TITLE
airbyte-ci: remove connector secrets hack for --is-local

### PR DESCRIPTION
**Pull request by postamar**
See https://github.com/airbytehq/airbyte/pull/33582#issuecomment-1878536246 for context. I tested that these changes worked by running airbyte-ci locally.

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/commands.py [hunk 1]
```diff
@@ -24,6 +23,8 @@ def deploy(ctx: click.Context) -> None:
 @deploy.command(cls=DaggerPipelineCommand, name="orchestrator", help="Deploy the metadata service orchestrator to production")
 @click.pass_context
 async def deploy_orchestrator(ctx: click.Context) -> None:
+    # Import locally to speed up CLI.
+    from pipelines.airbyte_ci.metadata.pipeline import run_metadata_orchestrator_deploy_pipeline
```
Comment (postamar) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/commands.py:27]: This shaves 4 seconds off of the execution time for `airbyte-ci --help`, bringing it down to under a second.

Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/commands.py:27]: Do you have a clue why the import is so slow?

File: airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py [hunk 2]
```diff
@@ -101,13 +101,8 @@ async def get_file_contents(container: Container, path: str) -> Optional[str]:
...
         Optional[str]: The file content if the file exists in the container, None otherwise.
     """
-    try:
-        return await container.file(path).contents()
-    except QueryError as e:
-        if "no such file or directory" not in str(e):
-            # this error could come from a network issue
-            raise
-    return None
+    nonexistence = await container.with_exec(["sh", "-c", f"[ -f '{path}' ] || echo '{path} does not exist'"]).stdout()
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py:104]: This consider that:
* `sh` is available in the container
* the container entrypoint is not overidden (you could set `skip_entrypoint=True` on the`with_exec`)

I'd find it more robust to only rely on Dagger's API to check file existence.
Maybe something like:
* Inferring the directory from the path
* Call `await container.directory(directory_path).entries()` and check if the file is in the returned list.

**Review by alafanechere — CHANGES_REQUESTED** [review]
As `get_file_contents` is made to be reused in different contexts on different containers I believe a solution to check file exsistence relying on dagger api instead of `sh` could be more stable.

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/commands.py (hunk 1 above)
Comment (postamar) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/commands.py:27]: It's not this import in particular, it's all of the transitive imports in `pipeline.py` and beyond.

File: airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py (hunk 2 above)
Comment (postamar) [reply airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py:104]: Yeah, I didn't think this one through. Good catch!

**Comment by postamar** [comment]
Thanks for the review, this is ready for another look.

File: airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py [hunk 3]
```diff
@@ -101,13 +101,10 @@ async def get_file_contents(container: Container, path: str) -> Optional[str]:
...
-        return await container.file(path).contents()
-    except QueryError as e:
-        if "no such file or directory" not in str(e):
-            # this error could come from a network issue
-            raise
-    return None
+    dir_name, file_name = os.path.split(path)
+    if file_name not in set(await container.directory(dir_name).entries()):
+        return None
+    return await container.file(path).contents()
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py:107]: The precondition here is that `path` is a file path and not a dir path, I'm not sure its enforced but I'm fine with it, Dagger will raise an exception if the path leads to a directory.

**Review by alafanechere — APPROVED** [review]
LGTM, thank you, our dagger  UI runs will be greener 🙏

File: airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py (hunk 3 above)
Comment (postamar) [reply airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py:107]: I think that raising an exception is fine in that case, `get_file_contents` really makes no sense if applied to a directory.

**Comment by postamar** [comment]
Bypassing branch protections owing to the failing formatting job, which is failing because it can't provision the right kind of runner or something.

**Pull request merged**